src/expiring_hash_map/tests.rs

```diff
@@ -0,0 +1,92 @@
+use super::*;
+use crate::runtime::Runtime;
+use futures::future::poll_fn;
+use futures_test::task::{new_count_waker, noop_context, panic_context};
+use std::time::{Duration, Instant};
+
+#[test]
+fn poll_does_not_return_ready_with_empty_map() {
+    let mut map = ExpiringHashMap::<String, String>::new();
+    let mut cx = noop_context();
+    assert!(map.poll_expired(&mut cx).is_pending());
+}
+
+#[test]
+fn poll_does_not_return_ready_with_empty_map_after_it_was_non_empty() {
+    let mut rt = Runtime::new().unwrap();
+    rt.block_on_std(async {
+        let mut map = ExpiringHashMap::<String, String>::new();
+
+        map.insert("key".to_owned(), "val".to_owned(), Duration::from_secs(1));
+        map.remove("key");
+
+        let mut cx = noop_context();
+        assert!(map.poll_expired(&mut cx).is_pending());
+    });
+}
+
+#[test]
+fn it_does_not_call_waker_if_polled_and_ready() {
+    let mut rt = Runtime::new().unwrap();
+    rt.block_on_std(async {
+        let mut map = ExpiringHashMap::<String, String>::new();
+
+        let a_minute_ago = Instant::now() - Duration::from_secs(60);
```

Comment:
We should also be using the built in test-util functionality to test timers, refer to the tokio time tests where we pause the time and advance it.

Reply:
I saw those at the `tokio` repo, but, unfortunately, everything is `pub(create)` there. It would be nice to use that indeed.

Reply:
I created https://github.com/tokio-rs/tokio/issues/2313

Reply:
What API's from clock do you need?

Reply:
Ideally all of them. I haven't looked thoroughly, but I think I'll need a clock-based `Instance` override there, and the calls to freeze time (`pause`, `advance`, etc).

My idea is that instead of practically waiting, we should freeze time, then shift time into the future. This, in turn, should trigger timer - and then we poll again, and the test should pass.

There should be a better way than doing `delay`s in tests, cause it's a problematic design for a whole lot of reasons. I'm already used to time freezing in more mature test ecosystems of other languages, so it really catches my attention. 😄